pyfolio/utils.py

```diff
@@ -259,3 +260,45 @@ def extract_rets_pos_txn_from_zipline(backtest):
     transactions.index.tz = None
 
     return returns, positions, transactions, gross_lev
+
+
+SETTINGS = {
+    # Settings dict to store functions/values that may
+    # need to be overridden depending on the users environment
+    'returns_func': default_returns_func
+}
+
+
+def register_return_func(func):
+    """
+    Registers the function that will be called for
+    retrieving returns data.
+
+    Parameters
+    ----------
+    func : function
+        A function that accepts a symbol and
+        returns a pandas.Series of asset returns.
+
+    Returns
+    -------
+    None
+    """
+    SETTINGS['returns_func'] = func
+
+
+def get_symbol_rets(symbol):
```

Comment:
We can probably merge it then, yeah.

On Tue, Jul 28, 2015 at 9:22 PM, David Edwards <notifications@github.com>
wrote:

> In pyfolio/utils.py
> <https://github.com/quantopian/pyfolio/pull/92#discussion_r35689042>:
>
> > +    retrieving returns data.
> > +
> > +    Parameters
> > +    ----------
> > +    func : function
> > +        A function that accepts a symbol and
> > +        returns a pandas.Series of asset returns.
> > +
> > +    Returns
> > +    -------
> > +    None
> > +    """
> > +    SETTINGS['returns_func'] = func
> > +
> > +
> > +def get_symbol_rets(symbol):
>
> Damn. I pushed with a partial before reading these last couple comments.
> Should I roll back to the single 'symbol' argument function and call it a
> day?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/quantopian/pyfolio/pull/92/files#r35689042>.
>